nixos/modules/programs/dolphin.nix

```diff
@@ -0,0 +1,78 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+
+let
+  cfg = config.programs.dolphin;
+in
+{
+  meta.maintainers = with maintainers; [ MakiseKurisu ];
+
+  # interface
+  options.programs.dolphin = {
+    enable = mkEnableOption (mdDoc ''
+      Dolphin, the KDE file manager.
+
+      This option will install additional depencencies to mimic the experience
+      of Dolphin in a normal KDE Plasma setup
+    '');
+
+    extraPackages = mkOption {
+      type = with types; listOf package;
+      default = with pkgs.libsForQt5; [
+        dolphin-plugins
+        kio-extras
+        kdegraphics-thumbnailers
+      ];
+      defaultText = literalExpression ''
+        with pkgs.libsForQt5; [
+          dolphin-plugins
+          kio-extras
+          kdegraphics-thumbnailers
+        ];
+      '';
+      description = lib.mdDoc ''
+        Extra packages that enhance Dolphin's functionality.
+
+        These packages are not strictly needed to resolve any UI inconsistency,
+        but are nice to have in most cases.
+      '';
+      example = literalExpression ''
+        with pkgs.libsForQt5; [
+          dolphin-plugins
+          kio-extras
+          kdegraphics-thumbnailers
+        ];
+      '';
+    };
+  };
+
+  # implementation
+  config = mkIf cfg.enable {
+
+    environment = {
+
+      # This is needed to have Konsole colorscheme files available in Dolphin
+      pathsToLink = [
+        "/share/konsole"
+      ];
+
+      # This is needed to select the Breeze theme
+      variables = {
+        QT_STYLE_OVERRIDE = "breeze";
```

Comment:
This seems aggressive and should at minimum be a `mkDefault`, but I think it is just the wrong way to do things. I am not sure the correct way, but it probably involves a config file on your specific machine.

Reply:
I now created a wrapper package to export this variable only for Dolphin. I also created 2 more options in case the user want to use other themes.